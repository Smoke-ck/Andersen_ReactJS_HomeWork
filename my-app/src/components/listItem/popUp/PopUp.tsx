import React, { FC } from 'react'
import { IToDos } from '../../../api'

type IPopup = {
    handleFavoriteToDo: (e:React.MouseEvent) => void,
    handleDoneToDo: (e:React.MouseEvent) => void,
    handleClick: () => void,
    handleToggleModal: () => void, 
    item: IToDos,
}

const PopUp:FC<IPopup> = ({handleFavoriteToDo,handleDoneToDo,handleClick,handleToggleModal,item}) => {
    return (
        <div>
            <button
                className="listItem__menu--button"
                onClick={(e: React.MouseEvent) => handleFavoriteToDo(e)}>
                {item.favorite ? 'Убрать из избранного ' : 'В избранно'}
            </button>
            <button
                className="listItem__menu--button"
                onClick={(e: React.MouseEvent) => handleDoneToDo(e)}>
                {item.completed ? 'Вернуть в работу' : 'Выполнено'}
            </button>
            <button
                className="listItem__menu--button"
                onClick={handleClick}> Редактировать
            </button>
            <button
                className="listItem__menu--button"
                onClick={handleToggleModal}> Удалить
            </button>
        </div>
    )
}

export default PopUp
