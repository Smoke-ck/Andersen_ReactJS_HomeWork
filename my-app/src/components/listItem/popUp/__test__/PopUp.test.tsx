import PopUp, { } from "../PopUp"
import { shallow, render, mount } from "enzyme";
import { shallowToJson } from 'enzyme-to-json';

const props = {
  handleFavoriteToDo: jest.fn(),
  handleDoneToDo: jest.fn(),
  handleClick: jest.fn(),
  handleToggleModal: jest.fn(),
  item: {
    title: "hi",
    completed: false,
    favorite: false,
    id: 50,
  }
}

describe('Selected component', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = mount(<PopUp{...props} />)
  })
  
  afterEach(() => {
    wrapper.unmount();
  })

  it("component should render correctly", () => {
    expect(wrapper).toBeTruthy();
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it("shoud render popup element", () => {
    const popup = wrapper.find('[className="popup"]');
    expect(popup).toHaveLength(1);
  })

  it("shoud render 4 button", () => {
    const buttons = wrapper.find('[className="listItem__menu--button"]');
    expect(buttons).toHaveLength(4);
  })

  it("shoud click for first button", () => {
    wrapper.find("button").at(0).simulate('click');
    expect(props.handleFavoriteToDo).toBeCalled();
  })

  it("shoud click for second button", () => {
    wrapper.find("button").at(1).simulate('click');
    expect(props.handleDoneToDo).toBeCalled();
  })

  it("shoud click for tird button", () => {
    wrapper.find("button").at(2).simulate('click');
    expect(props.handleClick).toBeCalled();
  })

  it("shoud click for fourth button", () => {
    wrapper.find("button").at(3).simulate('click');
    expect(props.handleToggleModal).toBeCalled();
  })
})


