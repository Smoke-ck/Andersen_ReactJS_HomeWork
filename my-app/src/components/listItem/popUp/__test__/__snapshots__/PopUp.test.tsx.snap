// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selected component component should render correctly 1`] = `
<PopUp
  handleClick={[MockFunction]}
  handleDoneToDo={[MockFunction]}
  handleFavoriteToDo={[MockFunction]}
  handleToggleModal={[MockFunction]}
  item={
    Object {
      "completed": false,
      "favorite": false,
      "id": 50,
      "title": "hi",
    }
  }
>
  <div
    className="popup"
  >
    <button
      className="listItem__menu--button"
      onClick={[Function]}
    >
      В избранно
    </button>
    <button
      className="listItem__menu--button"
      onClick={[Function]}
    >
      Выполнено
    </button>
    <button
      className="listItem__menu--button"
      onClick={[MockFunction]}
    >
       Редактировать
    </button>
    <button
      className="listItem__menu--button"
      onClick={[MockFunction]}
    >
       Удалить
    </button>
  </div>
</PopUp>
`;
